File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private boolean isSubtree = false;

    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        if (isSubtree) return true;
        
        if (root.val == subRoot.val) {
            isSubtree = isSameTree(root, subRoot);
        }
        if (root.left != null) {
            isSubtree(root.left, subRoot);
        }
        if (root.right != null) {
            isSubtree(root.right, subRoot);
        }
        return isSubtree;
    }

    private boolean isSameTree(TreeNode node, TreeNode subRoot) {
        if (subRoot == null && node == null) return true;
        if (subRoot == null || node == null) return false;
        if (subRoot.val != node.val) return false;

        return isSameTree(node.left, subRoot.left) && isSameTree(node.right, subRoot.right);
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root: return []

        q = deque([root])
        result = []

        while q:
            level = len(q)
            level_order = []
            for i in range(level):
                node = q.popleft()
                level_order.append(node.val)
                if node.left: q.append(node.left)
                if node.right: q.append(node.right)
            if level_order:
                result.append(level_order)
        
        return result